osu.Game/Overlays/ChatOverlayV2.cs

```diff
@@ -346,5 +386,19 @@ private void handleChatMessage(string message)
             else
                 channelManager.PostMessage(message);
         }
+
+        private void cycleChannel(int direction)
+        {
+            List<Channel> overlayChannels = filterToChatChannels(channelManager.JoinedChannels).ToList();
+
+            if (overlayChannels.Count < 2)
+                return;
+
+            int currentIndex = overlayChannels.IndexOf(currentChannel.Value);
+
+            currentChannel.Value = overlayChannels[(currentIndex + direction + overlayChannels.Count) % overlayChannels.Count];
+        }
+
+        private IEnumerable<Channel> filterToChatChannels(IEnumerable channels) => channels.Cast<Channel>().Where(c => c.Type != ChannelType.System);
```

Comment:
Is this change correct? It implies that channels like multiplayer channels should show up in the overlay?

Reply:
probably not, but should show that the function is not clear. i was changing it to fix traversal and things like announce channels not being considered.

it should be written to exclude rather than include, i think.